cmd/web3/main.go

```diff
@@ -1672,7 +1771,7 @@ func printReceiptDetails(r *web3.Receipt, myabi *abi.ABI) {
 	}
 }
 
-func UpgradeContract(ctx context.Context, rpcURL, privateKey, contractAddress, newTargetAddress string, amount int) {
+func UpgradeContract(ctx context.Context, rpcURL, privateKey, contractAddress, newTargetAddress string, amount *big.Int) {
```

Comment:
Sidebar: I had a thought about type aliasing `type Wei big.Int` or something, so that it would be clear in places like this which units are expected, and so that we could add our own methods for conversions instead of calling helpers.